src/order/compactly_generated.lean

```diff
@@ -231,4 +253,42 @@ begin
   exact λ x, ⟨{x}, ⟨λ x _, h x, Sup_singleton⟩⟩,
 end
 
+/-- A compact element k has the property that any b < k lies below a "maximal element below k",
+which is to say [⊥, k] is coatomic. -/
+theorem Iic_coatomic_of_compact_element (k : α) (hk : k ≠ ⊥) (h : is_compact_element k) :
+  is_coatomic (set.Iic k) :=
+begin
+  refine_struct { .. }, rintros ⟨b, hbk⟩,
+  -- The case b = k is exactly the trivial case b = ⊤ in Iic k.
+  by_cases hb : b = k, { apply or.inl, simp only [hb], refl, },
+  -- Otherwise, note b < k and handle the non-trivial case.
+  replace hb : b < k := lt_of_le_of_ne hbk hb, right,
+  -- A coatom lying above b in [⊥, k] is precisely a maximal element of [b, k).
+  suffices : ∃ (m : set.Ico b k), ∀ (x : set.Ico b k), m ≤ x → x = m,
+  { obtain ⟨⟨m, hm₁⟩, hmax⟩ := this,
+    let hm₂ := le_of_lt hm₁.right, use ⟨m, hm₂⟩,
+    refine ⟨⟨ne_of_lt hm₁.right, _⟩, hm₁.left⟩,
```

Comment:
You could also combine the `use` statement into your `refine`, and go straight to
```suggestion
    refine ⟨⟨m, le_of_lt hm₁.right⟩,⟨⟨ne_of_lt hm₁.right, _⟩, hm₁.left⟩⟩,
```